### PR TITLE
IndirectSimulations - Correlations

### LINKED ISSUE #17284: Indirect Simulations - Velocity and Angular correlations
Add algorithms that calculate the Angular and Velocity correlations, both cross correlations and auto correlations.
All four algorithms along with respective unit tests and documentation have been written by Maris Serzans (summer student at MSG)

### PR DESCRIPTION
**Pull request by louisemccann**
Add algorithms that calculate the Angular and Velocity correlations, both cross correlations and auto correlations.
All four algorithms along with respective unit tests and documentation have been written by Maris Serzans (summer student at MSG)
Algorithms found under Simulations.

**To test:**

Velocity: \olympic\babylon5\Public\Louise McCann\trajectories.nc
Timestep: 2.0
Angular: \olympic\babylon5\Public\Louise McCann\trajectory_methyliodide.nc
Timestep: 10.0, SpeciesOne: C, SpeciesTwo: I, SpeciesThree: H

Fixes #17284.

[RELEASE NOTES](https://github.com/mantidproject/mantid/blob/fcfb1122f8d0f58728f8241604d59234e4bd782e/docs/source/release/v3.8.0/indirect_inelastic.rst)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant documentation been added/updated?
- [x] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by SpencerHowells** [comment]
We use logger.information instead of print.

**Comment by raquelalvarezbanos** [comment]
Having a quick look I noticed that the new algorithms are missing a user example in the documentation. In Mantid we should have at least one user example for each algorithm.

On the other hand, to test the algorithms I ran the unit tests and noticed that they produced the figures that will be added to the documentation, so I think in this case it would make sense to make the python code that is ran by the tests visible to users as user examples in the documentation. @louisemccann I can help you with this if you don't know how to do it. It should be very easy...

File: Framework/PythonInterface/plugins/algorithms/AngularAutoCorrelationsSingleAxis.py [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+from mantid.kernel import *
+from mantid.api import *
+
+from scipy.io import netcdf
+import numpy as np
+import matplotlib.pyplot as plt
+import re
+import math
+import time
+import sys
```
Comment (raquelalvarezbanos) [inline Framework/PythonInterface/plugins/algorithms/AngularAutoCorrelationsSingleAxis.py]: Do we need `mantplotlib.pyplot`, `math` and `sys` for this algorithm?

File: Framework/PythonInterface/plugins/algorithms/AngularAutoCorrelationsTwoAxes.py [hunk 2]
```diff
@@ -0,0 +1,319 @@
+from mantid.simpleapi import *
+from mantid.kernel import *
+from mantid.api import *
+
+from scipy.io import netcdf
+import numpy as np
+import matplotlib.pyplot as plt
+import re
+import math
+import time
```
Comment (raquelalvarezbanos) [inline Framework/PythonInterface/plugins/algorithms/AngularAutoCorrelationsTwoAxes.py]: Same here: do we need `matplotlib` and `math`?

**Comment by raquelalvarezbanos** [comment]
Looks good to me. Four new algorithms (written by a summer student at MSG) to calculate angular and velocity correlations have been added, together with documentation and unit tests. :shipit:

**Pull request merged**